apps/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+project(iotivity-lite-apps)
+
+if(UNIX)
+    add_executable(simpleserver
+        ${PROJECT_SOURCE_DIR}/simpleserver.c
+        ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+    )
+    target_link_libraries(simpleserver
+        iotivity-server-client
+    )
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/simpleserver_creds)
+
+    add_executable(simpleserver_pki
+        ${PROJECT_SOURCE_DIR}/simpleserver_pki.c
+        ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+    )
+    target_link_libraries(simpleserver_pki
+        iotivity-server-client
+    )
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/simpleserver_pki_creds)
+
+    add_executable(simpleclient
+        ${PROJECT_SOURCE_DIR}/simpleclient.c
+        ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+    )
+    target_link_libraries(simpleclient
+        iotivity-server-client
+    )
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/simpleclient_creds)
+
+    if(EXISTS ${PROJECT_SOURCE_DIR}/device_builder_server.c)
+        add_executable(device_builder_server
+            ${PROJECT_SOURCE_DIR}/device_builder_server.c
+            ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+        )
+        target_link_libraries(device_builder_server
+            iotivity-server-client
+        )
+        file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/device_builder_server_creds)
+    endif()
+elseif(WIN32)
+    add_executable(simpleserver
+        ${PROJECT_SOURCE_DIR}/simpleserver_windows.c
+        ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+    )
+    target_link_libraries(simpleserver
+        iotivity-server-client
+    )
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/simpleserver_creds)
+
+    add_executable(simpleclient
+        ${PROJECT_SOURCE_DIR}/simpleclient_windows.c
+        ${iotivity-lite_SOURCE_DIR}/api/oc_introspection.c
+    )
+    target_link_libraries(simpleclient
+        iotivity-server-client
+    )
+    file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/simpleclient_creds)
+endif()
+
+# copy credentials, used by example applications.
+add_custom_target(copy-pki-certs ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+    ${PROJECT_SOURCE_DIR}/pki_certs
+    ${PROJECT_BINARY_DIR}/pki_certs
+)
```

Comment:
I don't think this does as intended. This adds the copy operation to the 'all' target so that when you type for instance ``make`` or ``make all``, this operation is executed. But if you type ``make simpleclient`` or any other specific target, it will not be.

The solution to this is one of:
- Just copy the entire directory at configure time (I think this makes the most sense and is simplest)
  - use ``execute_process`` for this
- Add this target as a dependency of every application target or the pki library (not recommended, custom targets are always out of date)
- use ``add_custom_command(TARGET <target_here> POST_BUILD COMMAND <command here>`` to run the command every time a certain target is built (such as one of the core libraries)

Reply:
The problem is that we want to make sure that the certs in the build dir update whenever the certs in the source dir update. I am not sure that `execute_process` accomplishes this. I don't really think we lose anything by copying the certs more frequently than strictly necessary. Thoughts?